internal/store/store.go

```diff
@@ -141,6 +141,15 @@ func (s *Store) Loop(ctx context.Context) error {
 	}
 	defer s.subscribers.TeardownAll(context.Background())
 
+	// Set up a defer handler, and make sure to unlock the state
```

Comment:
this bug has been driving me crazy!!! it's what makes the tests timeout on panic